packages/core/src/modules/discover-features/FeatureRegistry.ts

```diff
@@ -0,0 +1,30 @@
+import type { Feature, FeatureQuery } from './models'
+
+import { injectable } from '../../plugins'
+
+@injectable()
+class FeatureRegistry {
+  private features: Feature[] = []
+
+  public register<F extends Feature = Feature>(feature: F) {
+    this.features.push(feature)
+  }
+
+  public query(queries: FeatureQuery[]) {
```

Comment:
This might need some small documentation about the if statement and how the match * works. would save a lot of time :) when trying to understand the code.

Reply:
You are right. At least we need to add the reference to the RFC where these wildcards are introduced (https://github.com/hyperledger/aries-rfcs/tree/b3a3942ef052039e73cd23d847f42947f8287da2/features/0557-discover-features-v2#queries-message-type)